src/keycloak-theme/login/pages/login-reset-password/login-reset-password.component.html

```diff
@@ -1,69 +1,86 @@
+@let i18n = i18nService.i18n$ | async;
+@if (i18n) {
```

Comment:
Here there is conceptually no reason for i18n ever being undefined.  
While we're fetching the resources for a specific language we can already display the page in english (the english resources are bundled) and re-render the page once the locale are downloaded.

Reply:
it is the default behaviour of the async pipe. when you have something async like this observable it might be null due to the index signature of the transform method:
```export declare class AsyncPipe implements OnDestroy, PipeTransform {
    private _ref;
    private _latestValue;
    private markForCheckOnValueUpdate;
    private _subscription;
    private _obj;
    private _strategy;
    constructor(ref: ChangeDetectorRef);
    ngOnDestroy(): void;
    transform<T>(obj: Observable<T> | Subscribable<T> | Promise<T>): T | null;
    transform<T>(obj: null | undefined): null;
    transform<T>(obj: Observable<T> | Subscribable<T> | Promise<T> | null | undefined): T | null;
    private _subscribe;
    private _selectStrategy;
    private _dispose;
    private _updateLatestValue;
    static ɵfac: i0.ɵɵFactoryDeclaration<AsyncPipe, never>;
    static ɵpipe: i0.ɵɵPipeDeclaration<AsyncPipe, "async", true>;
}
```

of course if it is always defined the @if clause always returns true, so is only required for typescript strict null check

Reply:
I'm not so familiar with Angular but `AsyncPipe` dosen't seem to be the best abstraction here.  
We have a value from the get go, it can just be changed at most one time after a few milliseconds.

Reply:
it is the recommended way to resolve async content